pkg/infra/block_collector.go

```diff
@@ -18,54 +18,68 @@ type BlockCollector struct {
 	thresholdP, totalP int
 	totalTx            int
 	registry           map[uint64]int
+	addressBlock       map[uint64]map[string]bool
```

Comment:
i suppose only one of these two maps is sufficient?

also, with 16 peers and several millions of blocks, this map could take significant memory space. I think we do need gc here.